tsl/src/chunk_api.c

```diff
@@ -1673,13 +1675,37 @@ chunk_create_empty_table(PG_FUNCTION_ARGS)
 
 	ht = ts_hypertable_cache_get_entry(hcache, hypertable_relid, CACHE_FLAG_NONE);
 	Assert(ht != NULL);
-	check_privileges_for_creating_chunk(hypertable_relid);
+
+	/*
+	 * If the chunk is created in the internal schema, become the catalog
+	 * owner, otherwise become the hypertable owner
+	 */
+	if (strcmp(schema_name, INTERNAL_SCHEMA_NAME) == 0)
+		uid = ts_catalog_database_info_get()->owner_uid;
+	else
+	{
+		Relation rel;
+
+		rel = table_open(ht->main_table_relid, AccessShareLock);
+		uid = rel->rd_rel->relowner;
+		table_close(rel, AccessShareLock);
```

Comment:
Looks like the new code isn't hit in a test.  I guess the new chunks need to be created in non-internal schema for this code path to be evaluated. I think this can be done by setting `associated_schema_name` when creating the hypertable.